internal/search/doc/doc_test.go

```diff
@@ -0,0 +1,130 @@
+package doc
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestFieldLoadSaver(t *testing.T) {
+
+	commonTestCases := []KustomizationDocument{
+		{
+			identifiers:   []Atom{"namePrefix", "metadata.name", "kind"},
+			FilePath:      "some/path/kustomization.yaml",
+			RepositoryURL: "https://example.com/kustomize",
+			CreationTime:  time.Now(),
+			DocumentData: `
+namePrefix: dev-
+metadata:
+  name: app
+kind: Deployment
+`,
+		},
+	}
+
+	for _, test := range commonTestCases {
+		fields, metadata, err := test.Save()
+		if err != nil {
+			t.Errorf("Error calling Save(): %s\n", err)
+		}
+		doc := KustomizationDocument{}
+		err = doc.Load(fields, metadata)
```

Comment:
maybe throw in a test of how Load returns an error on cast failure.

Reply:
Great idea